t/helper/test-quote.c

```diff
@@ -0,0 +1,30 @@
+#include "test-tool.h"
+#include "quote.h"
+#include "strbuf.h"
+#include "string.h"
+
+int cmd__quote_buf_pretty(int argc, const char **argv)
+{
+	struct strbuf buf_payload = STRBUF_INIT;
+	char* nullString = 0;
+	char empty[5] = {'\0'};
+	
+	if (!strcmp(argv[1], "nullString")){
+		strbuf_addstr(&buf_payload, ".");
+		sq_quote_buf_pretty(&buf_payload, nullString);
+		printf("%s.\n", buf_payload.buf);
```

Comment:
This construction is a little confusing and might meet with a little resistance upstream.
I think my concern is that you're using the 2 dots as boundary markers to help make
the shell test script easier.  And that's fine.  I think moving the dots to the printf makes
it clearer that they are just that and not part of the buffer.
Maybe something like:

    sq_quote_buf_pretty(&buf_payload, whatever);
    printf(".%s.\n", buf_payload.buf);

Reply:
maybe also have a short comment to that effect.